### PR TITLE
bump module version to 4.0.0 (4003)

### PR DESCRIPTION
**Pull request by scott-brust**
Cherry pick `4.0.0` release changelog and version definitions into `develop` / `5.x` see [sc-110432](https://app.shortcut.com/particle/story/110432/cherry-pick-4-0-0-bump-commit-into-develop)

### CODE REVIEW

**Review by avtolstoy — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -136,6 +136,64 @@
...
+- [workbench] Remove redundant toolchain overrides [#2489](https://github.com/particle-iot/device-os/pull/2489)
+- [wifi] Add stubs to allow setting common country code and channel plan in P2 [#2473](https://github.com/particle-iot/device-os/pull/2473) [#2485](https://github.com/particle-iot/device-os/pull/2485)
+- [tests] Change PWM pins to better match e-series pinout for E SoM X. Fix other tests [#2463](https://github.com/particle-iot/device-os/pull/2463)
+- [tests] Misc changes to tests from 5.x [#2466](https://github.com/particle-iot/device-os/pull/2466)
+- [tests][esomx] disable RGB pwm channel(s) when running pwm tests [#2494](https://github.com/particle-iot/device-os/pull/2494)
+- Streamline `.bundleignore` between 5.x and 4.x codebases [#2496](https://github.com/particle-iot/device-os/pull/2496)
+- branch pattern changes for `test-build-system` [#2510](https://github.com/particle-iot/device-os/pull/2510)
+- [test] fixes race condition with `wiring/no_fixture` `TIME_17` [#2514](https://github.com/particle-iot/device-os/pull/2514)
+>>>>>>> cccaa6ca6 (bump module version to 4.0.0 (4003))
```
Comment (technobly) [inline CHANGELOG.md:195]: Nit: git foobar

File: system/inc/system_version.h [hunk 2]
```diff
@@ -191,6 +191,8 @@ extern "C" {
 #define SYSTEM_VERSION_v400ALPHA1   SYSTEM_VERSION_ALPHA(4, 0, 0, 1)
 #define SYSTEM_VERSION_v400ALPHA2   SYSTEM_VERSION_ALPHA(4, 0, 0, 2)
 #define SYSTEM_VERSION_v400BETA1   SYSTEM_VERSION_BETA(4, 0, 0, 1)
+#define SYSTEM_VERSION_v400         SYSTEM_VERSION_DEFAULT(4, 0, 0)
+// These are required for some of the dynalib checks
```
Comment (technobly) [inline system/inc/system_version.h:195]: Do we need this comment, and in this location?

**Review by technobly — APPROVED** [review]

File: system/inc/system_version.h (hunk 2 above)
Comment (scott-brust) [reply system/inc/system_version.h:195]: I dont know, it was in the 4.0.0 commit. It seems it refers to 5.0.0 alpha 1 and 2 versions, not sure if its relevant anymore

Comment (technobly) [reply system/inc/system_version.h]: Yeah we can safely remove that 👍🏼

**Pull request merged**